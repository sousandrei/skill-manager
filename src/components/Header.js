import React from 'react'
import PropTypes from 'prop-types'

const Header = ({
	name,
	addSkill,
	experience,
	handleChange
}) =>
	(
		<section className='header'>
			add your skills
			<input
				name='name'
				type='text'
				value={name}
				onChange={handleChange} />

			<select
				name='experience'
				value={experience}
				onChange={handleChange}>
				<option value='' disabled hidden>Experience</option>
				<option value='1 year'> {'<'} 1 year</option>
				<option value='1 - 3 years'>1 - 3 years</option>
				<option value='3 - 5 years'>3 - 5 years</option>
				<option value='5 - 7 years'>5 - 7 years</option>
				<option value='7+ years'>7+ years</option>
			</select>

			<button type='button' onClick={addSkill}>Click Me!</button>
		</section>
	)

Header.propTypes = {
	name: PropTypes.string,
	addSkill: PropTypes.func,
	experience: PropTypes.string,
	handleChange: PropTypes.func
}

export default Header

