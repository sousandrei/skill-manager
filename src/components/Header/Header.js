import React from 'react'
import PropTypes from 'prop-types'

const Header = ({
	name,
	addSkill,
	experience,
	handleChange
}) => {
	let invalid = false

	if (!name.length)
		invalid = true

	if (!experience.length)
		invalid = true

	return (
		<section className='header'>
			<input
				name='name'
				type='text'
				placeholder='Node JS, Postgres, React, etc.,'
				value={name}
				onChange={handleChange} />

			<div className='header__second-line'>
				<select
					name='experience'
					value={experience}
					onChange={handleChange}>
					<option value='' disabled hidden>Experience</option>
					<option value='< 1 year'> {'<'} 1 year</option>
					<option value='1 - 3 years'>1 - 3 years</option>
					<option value='3 - 5 years'>3 - 5 years</option>
					<option value='5 - 7 years'>5 - 7 years</option>
					<option value='7+ years'>7+ years</option>
				</select>

				<button
					type='button'
					disabled={invalid}
					onClick={addSkill}>ADD SKILLS</button>
			</div>
		</section>
	)
}

Header.propTypes = {
	name: PropTypes.string,
	addSkill: PropTypes.func,
	experience: PropTypes.string,
	handleChange: PropTypes.func
}

export default Header

