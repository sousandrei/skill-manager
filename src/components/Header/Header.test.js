import React from 'react'
import { shallow } from 'enzyme'

import Header from './Header'

test('should render header', () => {

	const wrapper = shallow(<Header
		name={''}
		experience={''} />)

	expect(wrapper).toMatchSnapshot()
})

