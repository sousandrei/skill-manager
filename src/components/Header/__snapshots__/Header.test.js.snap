// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header 1`] = `
<section
  className="header"
>
  <input
    name="name"
    placeholder="Node JS, Postgres, React, etc.,"
    type="text"
    value=""
  />
  <div
    className="header__second-line"
  >
    <select
      name="experience"
      value=""
    >
      <option
        disabled={true}
        hidden={true}
        value=""
      >
        Experience
      </option>
      <option
        value="< 1 year"
      >
         
        &lt;
         1 year
      </option>
      <option
        value="1 - 3 years"
      >
        1 - 3 years
      </option>
      <option
        value="3 - 5 years"
      >
        3 - 5 years
      </option>
      <option
        value="5 - 7 years"
      >
        5 - 7 years
      </option>
      <option
        value="7+ years"
      >
        7+ years
      </option>
    </select>
    <button
      disabled={true}
      type="button"
    >
      ADD SKILLS
    </button>
  </div>
</section>
`;
